storage/replica_raftstorage.go

```diff
@@ -187,18 +187,19 @@ func (r *Replica) LastIndex() (uint64, error) {
 // raftTruncatedState returns metadata about the log that preceded the first
 // current entry. This includes both entries that have been compacted away
 // and the dummy entries that make up the starting point of an empty log.
+// raftTruncatedState requires that the replica lock be held.
```

Comment:
The non-exported methods in this file that require the lock should have a `Locked` suffix.